src/providers/tfc/workspaceProvider.ts

```diff
@@ -185,13 +185,18 @@ export class WorkspaceTreeItem extends vscode.TreeItem {
       ? `$(source-control) [${this.attributes['vcs-repo-identifier']}](${this.attributes['vcs-repo']['repository-http-url']})`
       : '';
 
+    const statusIcon = GetRunStatusIcon(lastRun?.status ?? '');
+
+    const statusMsg = GetRunStatusMessage(lastRun?.status ?? '');
```

Comment:
Given that there can be a workspace without any (last) runs, what's the expected/intended behaviour there?

Reply:
No runs mean no status to report. TFC website does not show a status for workspaces with no runs.

Reply:
AFAICT the code here would produce empty strings which would below error out because we attempt to access `id` on empty string `$(${statusIcon.id})` wouldn't it?

TL;DR don't we need an extra conditional here somewhere?

Reply:
`GetRunStatusIcon` returns a `new vscode.ThemeIcon('indent')` as fallback, so I think everything should work

Reply:
True but `GetRunStatusMessage` will return empty string, so below we'd print out `Run Status: \n`, wouldn't we?

Reply:
Oh, now I see we print out the fallback icon first, then the message - sorry. That's fine then.

Reply:
That's true! We could hide `Run Status:` when the message is empty